src/App.js

```diff
@@ -7,35 +7,56 @@ import { HomeButton } from "./components/HomeButton";
 import { SingleRev } from "./components/SingleRev";
 import { UserContext } from "./contexts/UserContext";
 import { LoggedUser } from "./components/LoggedUser";
-
+import { CatFilters } from "./components/CatFilters";
+import { Categories } from "./components/Categories";
 
 function App() {
+  const [revList, setRevList] = useState([]);
   const [loading, setLoading] = useState(true);
   const [reviewId, setReviewId] = useState("");
-  const [username, setUsername] = useState("grumpy19")
+  const [username, setUsername] = useState("grumpy19");
+  const [innerUrl, setInnerUrl] = useState("");
```

Comment:
Have a look at using react router parametric endpoints and useParams to make the way state is handled cleaner.